kedro/ipython/__init__.py

```diff
@@ -170,7 +170,7 @@ def _resolve_project_path(
         ):
             project_path = local_namespace["context"].project_path
         else:
-            project_path = _find_kedro_project(Path.cwd())
+            project_path: Path | None = _find_kedro_project(Path.cwd())
```

Comment:
Without that, `mypy` complains that we assign whatever is returned from `_find_kedro_project`, which is `Path | None`, to `project_path` which is just `Path`